[Guest List and Network Highlight] Add header

mebarbosa opened the pull request:
## Description
- This PR adds the header to the collection
- Track header tapping event
- Add subscribe button 
- Design: [zfZ5UzmdLBk3k4CwaFIN3E-fi-3646_53376](https://href.li/?https://www.figma.com/file/zfZ5UzmdLBk3k4CwaFIN3E/?node-id=3646-53376)

## Testing Instructions
1. With `GUEST_LISTS_NETWORK_HIGHLIGHTS_REDESIGN` ON
2. Open Discover
3. Scroll until you see the Guest list, Guest Curator or Network highlight collection 
4. See that the collection shows a header only for the fist page ✅
5. Tap on the header
6. Ensure 🔵 Tracked: discover_list_collection_header_tapped, Properties: {"list_id":"ilst-name", ✅
7. Ensure the collection opens ✅ (same behavior as tapping on show all button)

## Screenshots or Screencast 


| Regular device | Large font size | Landscape |
|--------|--------|--------|
| ![Screenshot_20250319_123111](https://github.com/user-attachments/assets/7134bc58-44ca-460a-8be8-970830343b09) | ![Screenshot_20250319_123134](https://github.com/user-attachments/assets/229cbcd1-af3f-440a-a975-6e1dea91e866) | ![Screenshot_20250319_123155](https://github.com/user-attachments/assets/6c33ee32-52a8-41f0-a3d7-51e778312efc) | 

[Screen_recording_20250319_122915.webm](https://github.com/user-attachments/assets/b7c60cc1-f311-471e-b25a-52d84172d65b)



## Checklist
- [ ] If this is a user-facing change, I have added an entry in CHANGELOG.md
- [x] Ensure the linter passes (`./gradlew spotlessApply` to automatically apply formatting/linting)
- [x] I have considered whether it makes sense to add tests for my changes
- [x] All strings that need to be localized are in `modules/services/localization/src/main/res/values/strings.xml`
- [ ] Any jetpack compose components I added or changed are covered by compose previews
- [x] I have updated (or requested that someone edit) [the spreadsheet](https://docs.google.com/spreadsheets/d/107jqrutZhU0fVZJ19SBqxxVKbV2NWSdQC9MFYdLiAxc/edit?usp=sharing) to reflect any new or changed analytics.
 
#### I have tested any UI changes...
<!-- If this PR does not contain UI changes, ignore these items -->
- [x] with different themes
- [x] with a landscape orientation
- [x] with the device set to have a large display and font size
- [x] for accessibility with TalkBack

geekygecko commented on modules/features/discover/src/main/res/layout/collection_header.xml:
```diff
@@ -0,0 +1,63 @@
+<androidx.constraintlayout.widget.ConstraintLayout xmlns:android="http://schemas.android.com/apk/res/android"
+    xmlns:app="http://schemas.android.com/apk/res-auto"
+    xmlns:tools="http://schemas.android.com/tools"
+    android:layout_width="wrap_content"
+    android:layout_height="match_parent"
+    android:layout_marginVertical="8dp"
+    android:layout_marginLeft="8dp"
+    android:clickable="true"
+    android:focusable="true">
+
+    <com.google.android.material.card.MaterialCardView
+        android:layout_width="179dp"
+        android:layout_height="match_parent"
+        app:layout_constraintStart_toStartOf="parent"
+        app:layout_constraintTop_toTopOf="parent">
```
Could these be combined? I could be wrong but I don't think it needs a root view.

```suggestion
<com.google.android.material.card.MaterialCardView xmlns:android="http://schemas.android.com/apk/res/android"
    xmlns:tools="http://schemas.android.com/tools"
    android:layout_width="179dp"
    android:layout_height="match_parent"
    android:layout_marginVertical="8dp"
    android:layout_marginLeft="8dp"
    android:clickable="true"
    android:focusable="true">
```

geekygecko commented on modules/features/discover/src/main/res/layout/collection_header.xml:
```diff
@@ -0,0 +1,63 @@
+            android:layout_height="match_parent"
+            android:background="?attr/defaultArtwork"
+            android:scaleType="centerCrop"
+            tools:ignore="ContentDescription" />
```
On first load the view seems to larger than the row height. 
<img width="458" alt="Screenshot 2025-03-12 at 2 50 41 pm" src="https://github.com/user-attachments/assets/b718486f-b569-4bb3-b064-4f127c973040" />

geekygecko reviewed (commented):
Is it possible to get the following closer to the design:

1. The drop shadows are being cut off.
2. Can the subtitle have some more horizontal padding.
3. Can the two podcasts title and author be cut off similar to the design rather than wrapped?

![Screenshot_20250312_153458](https://github.com/user-attachments/assets/ac74a086-1333-4700-8b9a-0cd58367ce95)

<img width="361" alt="Screenshot 2025-03-12 at 3 48 10 pm" src="https://github.com/user-attachments/assets/74764688-d828-438a-b66b-a561b6cf3db1" />

geekygecko commented on modules/features/discover/src/main/java/au/com/shiftyjelly/pocketcasts/discover/view/CollectionListRowAdapter.kt:
```diff
@@ -62,31 +113,68 @@ internal class CollectionListRowAdapter(
-        val podcastSublist = getItem(position)
-        holder.bind(podcastSublist)
+    fun getPodcastsHeight(view: View): Int {
+        view.measure(0, 0)
```
The comment above was great in explaining the `modifiedList` logic. Could we do the same here, as I wasn't sure what this line did?

mebarbosa replied:
I missed this in my own review. I've added a comment for it

geekygecko commented on modules/features/discover/src/main/java/au/com/shiftyjelly/pocketcasts/discover/view/CollectionListRowAdapter.kt:
```diff
@@ -62,31 +113,68 @@ internal class CollectionListRowAdapter(
+                holder.binding.header.lblTitle.text = it.title
+                holder.binding.header.lblSubtitle.text = it.subtitle
+                holder.binding.header.imageHeader.load(it.imageUrl)
+                holder.binding.header.root.contentDescription =
+                    it.title + " " + holder.binding.root.context.getString(LR.string.discover_collection_header_content_description_suffix)
```
Should we be doing the concatenation in the strings file rather than here? Like the following:
```
<string name="podcast_next_episode_value">Next episode %s</string>
```

mebarbosa commented:
Hey @geekygecko I addressed all comments for this PR.

> Can the two podcasts title and author be cut off similar to the design rather than wrapped?

Regarding to this one, I asked about this here p1741786537792849/1741720182.869959-slack-C08HC6RUH40

geekygecko commented:
How's this change going? Is matching the design tricky?

geekygecko commented:
Did you see my message above? Is this PR ready to review?

mebarbosa commented:
@geekygecko Sorry. I miss the notification for this PR. Yes, it's ready for review

geekygecko commented:
The new paging is great! 

The only minor issues are:
- The drop shadow on white artwork doesn't look right. 
- @david-gonzalez-a8c there is quite a bit of padding between the artwork and the paging dots when compared with the featured component. Is this correct?

| | |
| --- | --- |
| ![Screenshot_20250325_154804](https://github.com/user-attachments/assets/2a4eac73-f615-49d6-9c03-11b65b409743) | ![Screenshot_20250325_154817](https://github.com/user-attachments/assets/5ecb53b4-d075-49af-bbdf-ef80d5d1abe7) |
